Java-Resources.md

```diff
@@ -0,0 +1,42 @@
+# Java Resources
+
+[Home](Java)
+
+Java is trademarked and licensed by Oracle. Most of the following are un-official resources, unless hosted by Oracle.
+
+## Tutorials
+
+- [Oracle Official Java Tutorials](http://docs.oracle.com/javase/tutorial/index.html)
+- [Thinking in Java](http://www.mindview.net/Books/TIJ/)
+- [Think Java](http://greenteapress.com/wp/think-java/)
+- [Effective Java by Joshua Bloch](https://github.com/HackathonHackers/programming-ebooks/blob/master/Java/Effective%20Java%20(2nd%20Edition).pdf)
+- [Jenkov's Java Tutorials](http://tutorials.jenkov.com/java/index.html)
+- [Mkyong's Java & Spring Tutorials](http://www.mkyong.com/)
+- [Vogella's Java Tutorials](http://www.vogella.com/tutorials/java.html)
+- [Java 8 Tutorial](https://github.com/winterbe/java8-tutorial)
+- [Better Java](https://github.com/cxxr/better-java)
+- [Java Programming notes by NTU](http://www3.ntu.edu.sg/home/ehchua/programming/index.html#Java)
+- [Princeton's Introduction to Programming in Java](http://introcs.cs.princeton.edu/java/home/)
+- [Introduction to Programming Using Java](http://math.hws.edu/javanotes/)
+- [Java Practices](http://javapractices.com/home/HomeAction.do)
+- [Java Design Patterns](https://github.com/iluwatar/java-design-patterns/)
+- [Algorithms in Java](https://github.com/pedrovgs/Algorithms)
+- [Spring Official Guides](https://spring.io/guides)
+
```

Comment:
I found these sites helpful too

http://www.learnjavaonline.org/ 
programmingbydoing.com
http://www.programcreek.com/java-tutorials/

jpassion.com is a very good resource, but there is a subscription fee of $139/year, so I don't know if that fits in with FCC or not. Their online codecamps are free to attend, paying members get access to materials
http://www.jpassion.com/portal/announcements/online-codecamps